src/bindings.rs

```diff
@@ -622,18 +627,23 @@ impl<'a> OwnedValueRef<'a> {
     pub fn new(context: &'a ContextWrapper, value: q::JSValue) -> Self {
         Self { context, value }
     }
+    pub fn new_dup(context: &'a ContextWrapper, value: q::JSValue) -> Self {
```

Comment:
No need for this method since it's covered by `clone()`.

Reply:
Well, not really, sometimes you get a value which you need to increment refcount on in order to "protect" it from GC, sometimes you don't
e.g. when calling get_property on an object qjs does not increment the refcount for you, when calling eval it does.. 
in quickjs_runtime i use OwnedValueRef(JSValueRef in my case) to always abstract recounting from the implementing developer... it is ofcourse also an option to leave that responsibility with the implementing developer, please let me know which way you want to go

Reply:
The point of `OwnedValueRef` is to remove the responsibilty for memory management from the caller.

You clone impl violates that.

I also wanted to add a `ValueRef` type that holds a reference to a `q::JSValue` and serves  the purpose you are intending, but I didn't have the need for it yet.

But all points in the code that have to deal with memory management should be marked `unsafe` to make you pay attention.
The memory leaks you are getting show that your current implementation is not ideal.

Ideally all code that has to respect memory management should live in `bindings.rs` and everything else should only use a memory safe wrapper.

Reply:
would it be acceptable to make the as_inner methods and such package private (pub(crate)) (e.g. never expose unsafe code to implementors)? that way we can use the OwnedValueRef as intended (abstract ref counting and such from implementors) but use it in utils like compile and others?

Reply:
Of should i make a JSValueRC which acts as an Rc<JSValue>.. An leave OwnedValueRef as it was?

Or should we just leave this and get busy with slimjs? :) fwiw, Awesome Idea!